code/game/objects/structures/xeno.dm

```diff
@@ -131,9 +131,17 @@
 		linked_carrier = builder
 	RegisterSignal(src, COMSIG_MOVABLE_SHUTTLE_CRUSH, .proc/shuttle_crush)
 
-/obj/effect/alien/resin/trap/proc/shuttle_crush() //No more trapping shuttles with huggies
+/obj/effect/alien/resin/trap/obj_destruction(damage_flag)
+	if(hugger && loc)
+		drop_hugger()
+
+	return ..()
+
+///No more trapping shuttles with huggies
+/obj/effect/alien/resin/trap/proc/shuttle_crush()
 	SIGNAL_HANDLER
 	QDEL_NULL(hugger)
+	hugger = null
```

Comment:
Unneeded

Reply:
This is literally what made the difference so it is very much needed.

Reply:
Qdel null Deletes and then nulls a variable I reccomend you go read what it does